code/game/objects/items/gamemode/survivor.dm

```diff
@@ -0,0 +1,340 @@
+    if(!user.get_active_held_item() && do_after(user, activation_time*1.2, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
+        reset_state()
+        user.put_in_hands(src)
+    . = ..()
+    
+/obj/item/laptop/rescue/attack_self(mob/user)
+    if(anchored)
+        to_chat(user, "<span class='warning'>It's already been anchored. Just leave it.</span>")
+        return
+
+    if(!ishuman(user)) 
+        return
+
+    if(!user.mind)
+        to_chat(user, "<span class='warning'>It doesn't seem to do anything for you.</span>")
+        return
+
+    var/area/A = get_area(user)
+    if(A && istype(A) && A.ceiling >= CEILING_METAL)
+        to_chat(user, "<span class='warning'>You have to be outside or under a glass ceiling to activate this.</span>")
+        return
+
+    user.visible_message("<span class='notice'>[user] starts setting up [src] on the ground.</span>",
+    "<span class='notice'>You start setting up [src] on the ground and inputting all the data it needs.</span>")
+
+    if(do_after(user, activation_time, TRUE, 5, BUSY_ICON_FRIENDLY,, TRUE))
+        log_game("[key_name(user)] placed \the [src] near [AREACOORD(A.loc)]")
+
+        user.transferItemToLoc(src, user.loc)
+        anchored = TRUE
+        w_class = 10
+        update_icon()
+        
+
+
+/obj/item/laptop/rescue/proc/reset_state()
+    icon_state = "tandy0"
+    w_class = initial(w_class)
+    anchored = FALSE
+    SetLuminosity(initial(luminosity))
+    update_icon()
+
+
+/obj/item/laptop/rescue/attack_alien(mob/living/carbon/Xenomorph/M)
+    if(anchored == FALSE)
+        to_chat(M, "<span class='warning'>It's already broken.</span>")
+        return
+        
+    M.animation_attack_on(src)
+    M.visible_message("<span class='danger'>\The [M] slices [src] apart!</span>", \
+        "<span class='danger'>You slice [src] apart!</span>", null, 5)
+    playsound(loc, "alien_claw_metal", 25, 1)
+
+    if (anchored == FALSE)
+        return
+
+    reset_state()
+   
+    if(prob(10))
+        new /obj/effect/decal/cleanable/blood/oil(src.loc)
+
+/obj/item/beacon/rescue
+    name = "Rescue beacon"
+    desc = "A rugged, glorified laser pointer capable of sending a beam into space. Activate and throw this to call for someone to rescue you."
+    icon_state = "motion0"
+    resistance_flags = UNACIDABLE|INDESTRUCTIBLE
+    w_class = WEIGHT_CLASS_BULKY
+    luminosity = 4 
+
+    var/activated = FALSE
+    var/activation_time = 10 SECONDS
+    var/icon_activated = "motion2"
+
+    var/distress_timer = 10 MINUTES
+    var/required_components = list(
+        /obj/item/cell,
+    )
+
+    // Requirements for nearby components that need setting up
+    var/scan_range = 2
+    var/required_nearby = list(
+        /obj/item/laptop/rescue,
+    )
+
+    // Contents list, for when it gets broken. We spit everything out again
+    var/internal_components = list()
+
+    // Custom health, It can't be destroyed but it can be stopped by reducing health to 0
+    var/max_hp = 100
```

Comment:
Use obj_integritiy for this

Reply:
Is there not other code that interacts with `obj_integritiy` I intentionally wanted to stay away from that.

Reply:
All objects should be handled through that

Reply:
The question is: does this represent object integrity or not? In this case it seems so.
It doesn't mean it needs to be destroyed once it reaches zero or lower. But if the way to lower this is through damage, then it should definitely be integrity.